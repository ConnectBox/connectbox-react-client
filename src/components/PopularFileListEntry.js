import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { mediaTypes } from '../utils'

class PopularFileListEntry extends Component {
    static propTypes = {
        file: PropTypes.object.isRequired,
        hideTopBorder: PropTypes.bool.isRequired,
    }

  render () {
    const { hideTopBorder, file } = this.props
    const popIndicatorElements = []
    for (let j = 0; j < file.score; j++) {
        popIndicatorElements.push(<span key={`score${j}`}><i className="fa fa-stop"></i></span>)
    }

    return (
        <tr>
            <td style={hideTopBorder ? {borderTop: 0} : {}}>
                <a style={{display: "inline-block"}} href={file.resource}>
                    <i className={`fa ${mediaTypes(file)}`}></i>
                    <span>&nbsp;{file.name}</span>
                </a>
                <span style={{display: "inline-block"}}>&nbsp;{popIndicatorElements}</span>
            </td>
        </tr>
    )
  }
}

export default PopularFileListEntry