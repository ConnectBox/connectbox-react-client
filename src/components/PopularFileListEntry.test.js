import React from 'react'
import PopularFileListEntry from './PopularFileListEntry'
import renderer from 'react-test-renderer'

test('Render PopularFileListEntry with hidden top border', () => {
  const iconMetadata = { ids: [], names: {}}
  const file = {type: 'file', ext: 'doc', name: 'Test File'}
  const component = renderer.create(
    <PopularFileListEntry
      contentRoot='/content'
      key='fooEntry0'
      file={file}
      hideTopBorder
      folderPath='foo'
      iconMetadata={iconMetadata} />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Render PopularFileListEntry without hidden top border', () => {
  const iconMetadata = { ids: [], names: {}}
  const file = {type: 'file', ext: 'doc', name: 'Test File'}
  const component = renderer.create(
    <PopularFileListEntry
      contentRoot='/content'
      key='fooEntry0'
      file={file}
      hideTopBorder={false}
      folderPath='foo'
      iconMetadata={iconMetadata} />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
