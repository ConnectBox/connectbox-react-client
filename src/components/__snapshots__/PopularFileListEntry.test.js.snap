// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render PopularFileListEntry with hidden top border 1`] = `
<tr>
  <td
    style={
      Object {
        "borderTop": 0,
      }
    }
  >
    <a
      href={undefined}
      style={
        Object {
          "display": "inline-block",
        }
      }
    >
      <i
        className="fa fa-file-text-o"
      />
      <span>
         
        Test File
      </span>
    </a>
    <span
      style={
        Object {
          "display": "inline-block",
        }
      }
    >
       
    </span>
  </td>
</tr>
`;

exports[`Render PopularFileListEntry without hidden top border 1`] = `
<tr>
  <td
    style={Object {}}
  >
    <a
      href={undefined}
      style={
        Object {
          "display": "inline-block",
        }
      }
    >
      <i
        className="fa fa-file-text-o"
      />
      <span>
         
        Test File
      </span>
    </a>
    <span
      style={
        Object {
          "display": "inline-block",
        }
      }
    >
       
    </span>
  </td>
</tr>
`;
